### PR TITLE
feat: add ml.llm.GeminiTextGenerator model

### PR DESCRIPTION
**Pull request by GarrettWu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by ashleyxuu — COMMENTED** [review]
Please add the GeminiTextGenerator model in the README.rst as well : https://github.com/googleapis/python-bigquery-dataframes/blob/main/README.rst?plain=1#L190

File: bigframes/ml/llm.py [hunk 1]
```diff
@@ -396,3 +398,175 @@ def to_gbq(
...
+        col_label = cast(blocks.Label, X.columns[0])
+        X = X.rename(columns={col_label: "prompt"})
+
+        options = {
+            "temperature": temperature,
+            "max_output_tokens": max_output_tokens,
+            "top_k": top_k,
+            "top_p": top_p,
+            "flatten_json_output": True,
+        }
```
Comment (ashleyxuu) [inline bigframes/ml/llm.py:547]: I believe we don't give Python users the option to specify this `flatten_json_output` as we return a DataFrame format? I guess it is only applicable in SQL interface?

File: docs/templates/toc.yml [hunk 2]
```diff
@@ -107,6 +107,8 @@
         uid: bigframes.ml.llm.PaLM2TextGenerator
       - name: PaLM2TextEmbeddingGenerator
         uid: bigframes.ml.llm.PaLM2TextEmbeddingGenerator
+      - name: GeminiTextGenerator
```
Comment (ashleyxuu) [inline docs/templates/toc.yml:110]: Should be sorted it in alphabetic order.

File: bigframes/ml/llm.py (hunk 1 above)
Comment (GarrettWu) [reply bigframes/ml/llm.py:547]: right. It isn't user-facing, but what we send to SQL. It is always True value.

File: docs/templates/toc.yml (hunk 2 above)
Comment (GarrettWu) [reply docs/templates/toc.yml]: Done

**Comment by GarrettWu** [comment]
> Please add the GeminiTextGenerator model in the README.rst as well : https://github.com/googleapis/python-bigquery-dataframes/blob/main/README.rst?plain=1#L190

Done

**Review by ashleyxuu — APPROVED** [review]

**Pull request merged**